alpaca/rest.go

```diff
@@ -314,7 +314,7 @@ type CloseAllPositionsRequest struct {
 }
 
 // CloseAllPositions liquidates all open positions at market price.
-func (c *Client) CloseAllPositions(req CloseAllPositionsRequest) ([]Order, error) {
+func (c *Client) CloseAllPositions(req CloseAllPositionsRequest) (closeAllPositionsSlice, error) {
```

Comment:
The error returned is wrapped for each symbol, as each position is returned a separate object and status

Reply:
Yes, I understand the response returned, but this function declaration doesn't necessarily need to change.

I am just suggesting that if we were to make a breaking change, at least it should be more intuitive to use the response as parsing of the `Body` feels like the job for the SDK.

And a simpler change for now if we don't want to make a breaking change would be to return an error if there are any errors maybe? otherwise return the orders as it does already

Reply:
If we do want to make a breaking change, then I think ideally the return is a map of symbols to either an `OrderEntity` or `APIError` but I can't come up with a good suggestion yet other than something like:

```
CloseAllPositions(req CloseAllPositionsRequest) (map[string]OrderEntity, map[string]APIError, error)
```

Reply:
I agree with @Neal: 
- best if we can maintain backward compatibility
- if that's not possible (though I think it is), at least be strongly typed (avoid `interface{}`)

Reply:
And I don't think we can introduce the update without breaking the existing method signature. I've updated the PR with this commit https://github.com/alpacahq/alpaca-trade-api-go/pull/282/commits/6c112431faaefb5f941f14084230bb6b3a8799bc

Reply:
@sai-g why can't we keep existing method signature and return an error if there are any errors otherwise return the orders as we discussed? I know it's not ideal at least we can avoid breaking change for now and I don't think it is very common for any liquidation to fail anyways.

And I'd update the godoc comment accordingly so it's clear what the method does.

Reply:
Sorry about the confusion!! I thought it was okay to introduce the breaking change for the functionality since it's not being utilized enough. Anyway, I made the update to not change the method signature in the recent commit https://github.com/alpacahq/alpaca-trade-api-go/pull/282/commits/2ef5813e6ae9dcb8a0700c6384cd5f6200806815